lib/WUI/random.cpp

```diff
@@ -0,0 +1,10 @@
+#include <device/peripherals.h>
+#include "random.h"
+
+bool random32bit(uint32_t *random) {
+
+    if (HAL_RNG_GenerateRandomNumber(&hrng, random) != HAL_OK) {
```

Comment:
At which occasion can this fail? And what do we actually _do_ about the failure?

Reply:
Supposedly it fail in interrupt, there is also a [version](https://github.com/prusa3d/Prusa-Firmware-Buddy/blob/857bcc542dd712c8da8e3ef454d25cb518f574f7/lib/Drivers/STM32F4xx_HAL_Driver/Src/stm32f4xx_hal_rng.c#L632l) that cannot fail, but instead returns 0, but I think that is probably more dangerous given our random nonce treatment. I think that returning unauthorized with nonce being 0 in the unlikely case this fails is ok. It will probably confuse the client a bit, but will not get any access to secret content :man_shrugging: .